webapp/src/App.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
```

Comment:
Why put the webapp in webapp/src/ instead of just webapp?

Reply:
The `src` is the unbundled source code. The actual code that should run on Docker is inside `/build`, which can be generated with `npm build`

Reply:
@Thann It might be that this setup of having the source and requiring a build is overkill for this project. A couple of js files and an index would maybe be ok.

Maybe this setup would make more sense for a production ready user-facing app, where optimizing the front-end and running tests before making builds is critical.

Reply:
Previously, running `node server --build` would build the webapp AND start the server.
Also previously, the "unbundled source code" lived in "webapp" and the resulting bundles lived in "dist".
Additionally building a docker image would compile the webapp, so the image is ready to use.
The strange thing here is that all of these *had* a convention that was ignored and re-implemented without a clear reason.

Reply:
This is not an actual request to change the code as it is, @Thann. It is just a practice of React.

After struggling with the project setup for a couple of hours I figured I would just focus in the React side.

So, this is not an actual Pull Request.

Reply:
Regarding the production-ready comment, the server deps don't take long to `npm install` and therefore wouldn't inhibit fontend testing. If it did, you could setup your CI to get around this issue in a couple of different ways. Also, wouldn't you want to run server tests along with the frontend tests anyway?

Reply:
I'm sorry you experienced difficulty in getting the project setup.  Part of the challenge is to *integrate* your webapp into the project, so we can see how you react to dealing with other peoples code and conventions. =]